File: backend_net/ChronosDtnNetApi/DTOs/TransactionRequest.cs
// Por que: Importa o namespace System.ComponentModel.DataAnnotations para validação de valores de entrada na controller.
using System.ComponentModel.DataAnnotations;

// Por que: Define o namespace DTOs para agrupar as classes de mapeamento de payloads HTTP.
namespace ChronosDtnNetApi.DTOs;

// Por que: Record imutável que representa os dados necessários para submeter uma transação financeira orbital.
public record TransactionRequest(
    // Por que: ID da conta bancária de origem que sofrerá o débito (deve ser preenchido).
    [Required(ErrorMessage = "Conta de origem é obrigatória")]
    long? SourceAccountId,

    // Por que: ID da conta bancária de destino que receberá os créditos (deve ser preenchido).
    [Required(ErrorMessage = "Conta de destino é obrigatória")]
    long? DestAccountId,

    // Por que: O valor transferido deve ser maior que zero (positivo) e é obrigatório.
    [Required(ErrorMessage = "Valor da transação é obrigatório")]
    [Range(0.01, double.MaxValue, ErrorMessage = "O valor da transação deve ser positivo")]
    decimal? Amount,

    // Por que: Nível de prioridade DTN do pacote (0 = Baixa, 1 = Média, 2 = Alta).
    [Required(ErrorMessage = "Prioridade da transação é obrigatória")]
    [Range(0, 2, ErrorMessage = "Prioridade permitida está entre 0 e 2")]
    int? Priority
);

File: backend_java/src/main/java/br/com/fiap/chronosdtn/dto/TransactionRequest.java
package br.com.fiap.chronosdtn.dto;

// Por que: Importa as anotações do Jakarta Validation para validação de payload no nível da Controller REST.
import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import java.math.BigDecimal;

// Por que: Cria um record imutável contendo as chaves de contas e valores da operação de crédito.
public record TransactionRequest(
    // Por que: Conta de débito é obrigatória.
    @NotNull(message = "Conta de origem é obrigatória")
    Long sourceAccountId,

    // Por que: Conta de crédito é obrigatória.
    @NotNull(message = "Conta de destino é obrigatória")
    Long destAccountId,

    // Por que: O valor transferido não pode ser nulo e deve ser maior que zero.
    @NotNull(message = "Valor da transação é obrigatório")
    @Positive(message = "O valor da transação deve ser positivo")
    BigDecimal amount,

    // Por que: Prioridade do pacote DTN (0 = Baixa, 1 = Média, 2 = Alta).
    @NotNull(message = "Prioridade da transação é obrigatória")
    @Min(value = 0, message = "Prioridade mínima permitida é 0")
    @Max(value = 2, message = "Prioridade máxima permitida é 2")
    Integer priority
) {
}
